modules/matterhorn-index-service/src/test/java/org/opencastproject/index/service/impl/IndexServiceImplTest.java

```diff
@@ -975,11 +1029,58 @@ public void updatePresenters() throws IOException, org.osgi.service.cm.Configura
     utcDate.setTime(start.getTime());
     RRule rRule = new RRule(generateRule(days, utcDate.get(Calendar.HOUR_OF_DAY), utcDate.get(Calendar.MINUTE)));
     IndexServiceImpl indexServiceImpl = new IndexServiceImpl();
-    return indexServiceImpl.calculatePeriods(start.getTime(), end.getTime(), duration, rRule, tz);
+    return calculatePeriods(rRule, start.getTime(), end.getTime(), duration, tz);
   }
 
   private String generateRule(String days, int hour, int minute) {
     return String.format("FREQ=WEEKLY;BYDAY=%s;BYHOUR=%d;BYMINUTE=%d", days, hour, minute);
   }
 
+  //NOTE: Do not modify this without making the same modifications to the copy of this method in SchedulerServiceImpl
+  //I would have moved this to an abstract class in the scheduler-api bundle, but that would introduce a circular dependency :(
+  public List<Period> calculatePeriods(RRule rrule, Date start, Date end, long duration, TimeZone tz) {
```

Comment:
I want so badly to move this to an AbstractSchedulerService where it belongs, but that class can't live in scheduler-api without introducing a circular dependency with message-broker-api.  Improvements or suggestions welcome.